api-registration/server/index.js

```diff
@@ -0,0 +1,68 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  argon2.hash(password)
+    .then(password => {
+      const sql = `
+    insert into "users" ("username", "hashedPassword")
+    values ($1, $2)
+    returning "userId", "username", "createdAt"
+  `;
+
+      const param = [username, password];
+
+      db.query(sql, param)
+        .then(result => {
+          const [registration] = result.rows;
+          res.status(201).json(registration);
+        })
+        .catch(err => next(err));
+    })
+    .catch(err => {
+      console.error(err);
```

Comment:
Here you should also be calling `next(err)`, you also don't really need the `console.error` here, the error middleware will log out the error for you already.